src/common/core/corerel.js

```diff
@@ -171,17 +137,73 @@ define([
                 return false;
             }
         };
+
+        this.persist = function (node) {
+            var persisted = innerCore.persist(node);
+
+            storeNewInverseOverlays(self.getRoot(node));
+
+            return persisted;
+        };
         //</editor-fold>
 
         //<editor-fold=Added Methods>
+        this.getInverseOverlayOfNode = function (node) {
+            var hash = self.getHash(node),
+                inverseOverlays = {},
+                overlay,
+                source,
+                name,
+                target;
+
+            // If the node already has inverse computed we return that
+            if (node.inverseOverlays) {
+                return node.inverseOverlays;
+            }
+
+            // If we find it in the cache we set that and use it
+            if (hash) {
+                inverseOverlays = self._inverseCache.getItem(hash);
+                if (inverseOverlays) {
+                    node.inverseOverlays = inverseOverlays;
+                    return node.inverseOverlays;
+                }
+            }
+
+            // Otherwise we have to compute it
+            overlay = self.getProperty(node, CONSTANTS.OVERLAYS_PROPERTY);
+            overlay = overlay || {};
+            inverseOverlays = {};
+            for (source in overlay) {
+                for (name in overlay[source]) {
+                    target = overlay[source][name];
+                    inverseOverlays[target] = inverseOverlays[target] || {};
+                    inverseOverlays[target][name] = inverseOverlays[target][name] || [];
+                    inverseOverlays[target][name].push(source);
+                }
+            }
+
+            // If it is an unmodified node, we can store the inverse, otherwise it still can change
+            if (hash) {
+                self._inverseCache.setItem(hash, inverseOverlays);
+            }
+
+            node.inverseOverlays = inverseOverlays;
+
+            return node.inverseOverlays;
+
+        };
+
         this.isPointerName = function (name) {
             ASSERT(typeof name === 'string');
             //TODO this is needed as now we work with modified data as well
             if (name === CONSTANTS.MUTABLE_PROPERTY) {
                 return false;
             }
-            return name.slice(-CONSTANTS.COLLECTION_NAME_SUFFIX.length) !==
-                   CONSTANTS.COLLECTION_NAME_SUFFIX;
+            // return name.slice(-CONSTANTS.COLLECTION_NAME_SUFFIX.length) !==
+            //     CONSTANTS.COLLECTION_NAME_SUFFIX;
+
+            return true;
```

Comment:
After this change, the 'ptr-inv' can be a regular pointer name. That is why I have changed the function. Though we are not yet propagated this to the UI, so we can keep the check as it was.